src/main/kotlin/com/shankhadeepghoshal/kotlinfinaltask/controller/Controller.kt

```diff
@@ -0,0 +1,23 @@
+package com.shankhadeepghoshal.kotlinfinaltask.controller
+
+import com.shankhadeepghoshal.kotlinfinaltask.pojo.Dog
+import com.shankhadeepghoshal.kotlinfinaltask.service.DogsApiService
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class Controller(private val dogsApiService: DogsApiService) {
+    @GetMapping(path = ["/dogs"], produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE])
```

Comment:
Content type "application/octet-stream" is used for binary files. This endpoint is broken.